weave-js/src/components/Panel2/ChildPanelExportReport/ChildPanelExportReport.tsx

```diff
@@ -43,24 +46,67 @@ export const ChildPanelExportReport = ({
     null
   );
 
-  const selectedReportQuery = useQuery(GET_REPORT, {
+  const isNewReportSelected = isNewReportOption(selectedReport);
+
+  const [upsertReport, upsertReportResult] = useMutation(UPSERT_REPORT);
+  const reportQueryResult = useQuery(GET_REPORT, {
     variables: {id: selectedReport?.id ?? ''},
     skip: !selectedReport || isNewReportOption(selectedReport),
   });
 
-  const onAddPanel = () => {
-    const selectedReportDetails = selectedReportQuery.data?.view;
+  const isAddPanelDisabled =
+    !selectedEntity ||
+    !selectedReport ||
+    !selectedProject ||
+    reportQueryResult.loading ||
+    reportQueryResult.error != null ||
+    upsertReportResult.loading;
+
+  const onAddPanel = async () => {
+    if (isAddPanelDisabled) {
+      return;
+    }
+
     const slateNode = computeReportSlateNode(fullConfig, selectedPath);
-    // TODO - this will be replaced with correct add panel implementation later on
-    alert(`
-      Report id: ${selectedReport?.id}
-      Report name: ${selectedReport?.name}
-      Entity name: ${selectedEntity?.name}
-      Project name: ${selectedProject?.name}
-      Slate node: logged in dev console
-    `);
-    console.log('Selected report details:', selectedReportDetails);
-    console.log('Slate node to be added:', slateNode);
+    const entityName = selectedEntity.name;
+    const projectName = selectedProject.name;
+    let upsertBody;
+
+    if (isNewReportSelected) {
+      // https://wandb.atlassian.net/browse/WB-15495
+      alert('TODO: handle new report option');
+      return;
+    } else {
+      const publishedReport = reportQueryResult.data?.view;
+      const drafts = publishedReport?.children?.edges.map(({node}) => node);
+      const viewerDraft = drafts?.find(draft => draft?.user?.id === viewer.id);
+      if (viewerDraft) {
+        const spec = JSON.parse(viewerDraft.spec);
+        spec.blocks.push(slateNode);
+        upsertBody = {
+          id: viewerDraft.id,
+          spec: JSON.stringify(spec),
+        };
+      } else {
+        // https://wandb.atlassian.net/browse/WB-15494
+        alert('TODO: handle report without existing draft');
+        return;
+      }
+    }
+
+    const result = await upsertReport({variables: upsertBody});
+    const upsertedDraft = result.data?.upsertView?.view!;
+    const reportDraftPath = Urls.reportEdit({
+      entityName,
+      projectName,
+      reportID: upsertedDraft.id,
+      reportName: upsertedDraft.displayName ?? '',
+    });
+
+    // Should set hostname per env (e.g. dev, local, saas), presumably with urlPrefixed?
+    // eslint-disable-next-line wandb/no-unprefixed-urls
+    window.open(`https://wandb.ai/${reportDraftPath}`, '_blank');
```

Comment:
Not sure about hard-coded the https://wandb.ai portion, but I just copied what's being done here. @tssweeney -- thoughts?

Reply:
ahh, you want to do `urlPrefixed`

Reply:
actually, this just came up in standup. @vanpelt mentioned that you can use `window.WEAVE_CONFIG.WANDB_BASE_URL` and chop off the `api.` portion of the host if it exists

Reply:
Ah ok, will do that! TY

Reply:
@tssweeney updated! does [this](https://github.com/wandb/weave/pull/699/files#diff-a26f97625de2cca0b8eea5a13411a4d9ce5292a24c1a59df7206c350a58561a1R52) look appropriate?